docs/writing-rules/pattern-syntax.md

```diff
@@ -407,6 +407,9 @@ The deep expression operator works in:
 
 ## Statements types
 
+Many programming languages differentiate between expressions and statements. Those are 2 different things. Expressions can appear inside if conditions, in function call arguments, etc. Statements can not appear everywhere; they are sequence of operations (with ; as a separator/terminator) or special control flow constructs (if, while, etc.).
+
+
 Semgrep handles statements differently from other expressions like imports. For example, the pattern `foo` will match these statements:
```

Comment:
Actually import are not really expressions

Reply:
maybe "from other expressions like additions or function calls." ?

Reply:
actually the whole paragraph confuses statements and expressions I think, partly because in Python you don't need the semicolon.
The 3 things below are really expressions, not statements. 
They are statements when they are used in a sequence of operations, but bar + foo is weird to use in a sequence (unless you put a 'return' before). cc @mschwager who might have written that.